Feat/error state mobx

matthew-garrett opened the pull request.

jeffywu commented on apps/web/src/containers/App/RouteContainer.tsx:
```diff
@@ -22,4 +23,4 @@ const RouteContainer = ({ children }: RouteContainerProps) => {
 };
 
-export default RouteContainer;
+export default observer(RouteContainer);
```
Is adding observer at this level going to cause issues?

matthew-garrett committed:
Feat/arb points mobx (#1062)

* fix: refactor configuration worker to use r2 (#1017)

* fix: refactor configuration worker to use r2

* refactor: durable objects

* refactor: configuration registry

* fix: handle registry data

* fix: remove json parsing

* fix: handle views

* fix: put all registries in one place

* fix: remove other registries

* fix: adding view refresh to registry

* refactor: remove durable objects

* fix: remove data and add to cron jobs

* fix: change cache hostname

* fix: api env

* fix: account cache requests

* fix: update api wrangler

* fix: adding gnosis safe to onboard context (#1025)

* Build/fix web build (#1029)

* build: removes unused property (#1019)

* fix: don't log error on failed tx on first attempt (#1020)

* Bugfix/my breakdown multipliers (#1021)

* build: fixes points multiplier values

* build: fixes destructuring

* Bugfix/update liquidator (#1022)

* fix: increase threshold

* fix: update zero ex limit

* fix: revert dust threshold

* feat: update arb reinvestment for week 5/8 (#1023)

* feat: update mainnet owner in vault-apy config (#1024)

* Adding Next.js Shell App (#950)

* fix: adding analytics shell

* build: adds analytics app wip

* build: creates custom package.json

* fix: remove unused file

* fix: adding to nxignore

* fix: adding cache to nx

* build: adding common-icons dependency

* build: adds missing files

* fix: type errors

* build: fixes analytics build errors

* fix: build issues

* fix: adding cloudflare types

* build: add actions

* fix: update name

---------

Co-authored-by: matthew-garrett <matthew.garrett252@gmail.com>

* build: update yarn lock

* build: update workspace exclusion

* fix: remove dependency

* build: update build script

* fix: remove dependency

* build: move analytics to gcp folder

* build: reset yarn lock

* fix: debug lockfile

* fix: update yarn lock and workflow

* fix: contract build changes

* fix: update mui and remove unstyled

* fix: update packages and webpack config

* fix: update select dropdown

* build: fixes note icon bug

---------

Co-authored-by: matthew-garrett <98406539+matthew-garrett@users.noreply.github.com>
Co-authored-by: Sime <sbuljac@gmail.com>
Co-authored-by: matthew-garrett <matthew.garrett252@gmail.com>

* fix: update yarn lock

* fix: updated yarn again

* fix: update git workflow

* fix: update yarn lockfile again

* fix: node and yarn versions

* fix: node and yarn versions

* fix: update yarn.lock again

* fix: update registry url

* fix: asset select dropdown width (#1035)

* refactor: React 18 types, React Router 6 (#1032)

* build: moves hooks in app and app layout to components

* fix: refactor app to react 18

* build: updating packages to react-router 6

* refactor: updating components

* fix: switch to react router 6 and react 18 types

* fix: remove styled components and other ts fixes

---------

Co-authored-by: Jeff Wu <jeff@notional.finance>

* Feat/analytics dropdown (#1030)

* Feat/vault reinvestment date (#1027)

* build: adds vault reinvestment date to vault holdings table

* build: adds useEffect back into the table hook

* build: addresses comments

* build: updates refresh timing of registry (#1034)

* feat: updat ARB investments (#1036)

Co-authored-by: Sime Buljac <sime@Sims-MBP.localdomain>

* build: adds analytics network dropdown adds vault accounts page and menue options

build: removes const cast types

build: fix

* build: addresses comments

* fix: util package reference

---------

Co-authored-by: Sime <sbuljac@gmail.com>
Co-authored-by: Sime Buljac <sime@Sims-MBP.localdomain>
Co-authored-by: Jeff Wu <jeff@notional.finance>

* Bugfix/react router bugs (#1041)

* fix: roll debt and convert asset routes

* fix: route element wrapping

* fix: janky fix to router options

* fix: borrow terms

* fix: silence spindl unless it exists

* fix: default network bug

* fix: remove log

* Refactor/mobx state management (#1042)

* fix: add mobx, remove tests

* refactor: first pass at refactoring

* refactor: remove unused side drawers

* refactor: side drawer refactor

* fix: adding linting

* fix: update eslint

* fix: disable mobx warning

* fix: move side drawer manager

* build: fixes network being undefined when clicking launch app (#1043)

* build: adds mst app state with baseCurrency and themeVariant (#1044)

* build: adds mst app state with baseCurrency and themeVariant

* build: addresses comments

* build: import clean up

* fix: redner issue with leveraged liquidity (#1048)

* build: adds hero stats to mobx (#1049)

build: removes old import

build: addresses comments

* build: refactors side drawer headers (#1053)

* build: arb points mobx wip

* build: adds feature contexts for arb points

* build: addresses comments

* build: adds id

---------

Co-authored-by: Jeff Wu <jeff@notional.finance>
Co-authored-by: Sime <sbuljac@gmail.com>
Co-authored-by: Sime Buljac <sime@Sims-MBP.localdomain>
fix: refactor configuration worker to use r2 (#1017)

* fix: refactor configuration worker to use r2

* refactor: durable objects

* refactor: configuration registry

* fix: handle registry data

* fix: remove json parsing

* fix: handle views

* fix: put all registries in one place

* fix: remove other registries

* fix: adding view refresh to registry

* refactor: remove durable objects

* fix: remove data and add to cron jobs

* fix: change cache hostname

* fix: api env

* fix: account cache requests

* fix: update api wrangler

fix: adding gnosis safe to onboard context (#1025)

Build/fix web build (#1029)

* build: removes unused property (#1019)

* fix: don't log error on failed tx on first attempt (#1020)

* Bugfix/my breakdown multipliers (#1021)

* build: fixes points multiplier values

* build: fixes destructuring

* Bugfix/update liquidator (#1022)

* fix: increase threshold

* fix: update zero ex limit

* fix: revert dust threshold

* feat: update arb reinvestment for week 5/8 (#1023)

* feat: update mainnet owner in vault-apy config (#1024)

* Adding Next.js Shell App (#950)

* fix: adding analytics shell

* build: adds analytics app wip

* build: creates custom package.json

* fix: remove unused file

* fix: adding to nxignore

* fix: adding cache to nx

* build: adding common-icons dependency

* build: adds missing files

* fix: type errors

* build: fixes analytics build errors

* fix: build issues

* fix: adding cloudflare types

* build: add actions

* fix: update name

---------

Co-authored-by: matthew-garrett <matthew.garrett252@gmail.com>

* build: update yarn lock

* build: update workspace exclusion

* fix: remove dependency

* build: update build script

* fix: remove dependency

* build: move analytics to gcp folder

* build: reset yarn lock

* fix: debug lockfile

* fix: update yarn lock and workflow

* fix: contract build changes

* fix: update mui and remove unstyled

* fix: update packages and webpack config

* fix: update select dropdown

* build: fixes note icon bug

---------

Co-authored-by: matthew-garrett <98406539+matthew-garrett@users.noreply.github.com>
Co-authored-by: Sime <sbuljac@gmail.com>
Co-authored-by: matthew-garrett <matthew.garrett252@gmail.com>

fix: update yarn lock

fix: node and yarn versions

fix: node and yarn versions

refactor: React 18 types, React Router 6 (#1032)

* build: moves hooks in app and app layout to components

* fix: refactor app to react 18

* build: updating packages to react-router 6

* refactor: updating components

* fix: switch to react router 6 and react 18 types

* fix: remove styled components and other ts fixes

---------

Co-authored-by: Jeff Wu <jeff@notional.finance>

Feat/analytics dropdown (#1030)

* Feat/vault reinvestment date (#1027)

* build: adds vault reinvestment date to vault holdings table

* build: adds useEffect back into the table hook

* build: addresses comments

* build: updates refresh timing of registry (#1034)

* feat: updat ARB investments (#1036)

Co-authored-by: Sime Buljac <sime@Sims-MBP.localdomain>

* build: adds analytics network dropdown adds vault accounts page and menue options

build: removes const cast types

build: fix

* build: addresses comments

* fix: util package reference

---------

Co-authored-by: Sime <sbuljac@gmail.com>
Co-authored-by: Sime Buljac <sime@Sims-MBP.localdomain>
Co-authored-by: Jeff Wu <jeff@notional.finance>

Bugfix/react router bugs (#1041)

* fix: roll debt and convert asset routes

* fix: route element wrapping

* fix: janky fix to router options

* fix: borrow terms

* fix: silence spindl unless it exists

* fix: default network bug

Refactor/mobx state management (#1042)

* fix: add mobx, remove tests

* refactor: first pass at refactoring

* refactor: remove unused side drawers

* refactor: side drawer refactor

* fix: adding linting

* fix: update eslint

* fix: disable mobx warning

* fix: move side drawer manager
fix: redner issue with leveraged liquidity (#1048)

build: arb points mobx wip

build: adds feature contexts for arb points

build: addresses comments

build: adds id

build: moves app store to context

build: wip app context

build: moves signer to useTransaction hook

build: fixes theme changes and rebase

build: addresses comments

build: moves notional errors to mobx

build: fixes rebase issue

build: fixes

build: fixes rebase issues
